src/main/java/net/fortuna/ical4j/model/DateTime.java

```diff
@@ -529,7 +529,7 @@ private DateFormatCache(DateFormat dateFormat) {
 			this.templateFormat = dateFormat;
 		}
 
-		public DateFormat get() {
+		public synchronized DateFormat get() {
```

Comment:
The call to `threadMap.put(Thread.currentThread(), dateFormat)` would also have to be called while synchronized on the same lock.  The only thing you can reasonably pull out of the synchronized scope is the call to `templateFormat.clone()` and I don't think there's much benefit there.